analyses/fusion-summary/01-fusion-summary.Rmd

```diff
@@ -203,9 +232,133 @@ allFuseEwing %>%
+  pull(Gene_Symbol)
+
+lgatFuses_df <- lgatFuses_df %>% 
+  dplyr::mutate(Gene1A_anno = case_when(Gene1A %in% kinases ~ "Kinase"), 
+                Gene1B_anno = case_when(Gene1B %in% kinases ~ "Kinase")) %>%
+   replace_na(list(Gene1A_anno = "Non-Kinase", Gene1B_anno = "Non-Kinase"))
+
+# Only pull fusions that do not contain kinase genes, as ones with kinases will be dealt with sepaately later
+nonkinase_lgatFuses <- lgatFuses_df %>%
+  filter(Gene1A_anno == "Non-Kinase" & Gene1B_anno == "Non-Kinase") %>%
+  pull(lgatFuses)
+
+# Identify non-kinase genes in LGAT goi list
+nonkinase_lgatGenes <- setdiff(lgatGenes, kinases)
+
+# Pull LGAT non-kinase fusions
+nonkinaseLGAT <- filterFusion(df = putativeOncogenicDF,
+                             fuses = nonkinase_lgatFuses,
+                             genes = nonkinase_lgatGenes) %>%
+  distinct()
+```
+
+Next, collect fusions which contain 3' kinases which are in-frame and retain the kinase domain. 
+Keep these for the final output file.
+```{r}
+three_prime_kinase_inframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type == "in-frame" & DomainRetainedGene1B == "Yes") %>%
+  select(Sample, FusionName, Gene1A, Gene1B) %>%
+  distinct()
+
+# Are there any fusions that are in-frame, but do not retain the kinase domain? Do they have in-frame fusions retaining the kinase domain in the same fusion?
+three_prime_kinase_outframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type == "in-frame" & DomainRetainedGene1B == "No") %>%
+  select(Sample, FusionName) %>%
+  distinct()
+
+three_prime_kinase_outframe
+```
+
+Let's look at these just to be sure the results are as expected.
+```{r}
+# `BS_KE56MMY0 ARHGEF2--NTRK1` one does and will be captured in the `three_prime_kinase_inframe` list, but `BS_B1C6GZ84 CHIC2--PDGFRA2` does not retain the kinase domain, so we do not want to add it.
+intersect(three_prime_kinase_outframe[,c("Sample", "FusionName")], three_prime_kinase_inframe[,c("Sample", "FusionName")])
+```
+
+```{r}
+# Which fusions are not in-frame?
+three_prime_kinase_outframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type != "in-frame") %>%
+  distinct()
+# Check they don't retain the kinase domain. They do not, so leave out.
+table(three_prime_kinase_outframe$DomainRetainedGene1B)
+```
+
+Next, filter all fusions for 5' kinase fusions which have reciprocal fusions and retain the kinase domain.
+Keep these for the final output file.
+```{r}
+# Keep those with kinase domain retained and fusion in-frame - keep this list
+five_prime_domain_intact <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1A_anno) & DomainRetainedGene1A == "Yes" & Fusion_Type == "in-frame") %>%
+  select(Sample, FusionName) %>%
+  distinct()
+```
+
+Next, filter all fusions for 5' kinase fusions which have lost the kinase domain for reciprocal fusions which have a kinase domain intact and are in-frame.
+Keep these for the final output file.
+```{r}
```

Comment:
I'm thinking this chunk can still be simplified. In particular, I'm noticing that `five_prime_domain_lost` is from `allFuseLGAT` and `five_prime_reciprocals` is from `five_prime_domain_lost` but then we are adding back info from `allFuseLGAT`.

So basically both `five_prime_domain_lost` and `five_prime_reciprocals` are based on `allFuseLGAT` but we have extra manipulation steps. I think we can do this more directly.

Reply:
As I'm trying to make this shorter, I'm a bit confused on these steps. 

I tried to boil it down to what looks like the essential steps to get `five_prime_kinase_keep`, but as of now it does not get the same result. I think part of this is I still don't really understand all that is happening in this chunk. Can you take a look at my attempt as well evaluating your original code to see how we can make these steps more efficient and clear? 

In particular I know the differences have to do with the paired up  "select() then distinct()" steps that are done a few timees, but its unclear to me why each of these "select() then distinct()" steps are necessary. 

```
five_prime_kinase_keep <- allFuseLGAT %>%
  # Make reciprocal variable
  mutate(reciprocal = paste(Gene1B, Gene1A, sep ="--")) %>%
  filter(# Keep the in-frame reciprocals which have the kinase domain in tact.
        grepl("Kinase", Gene1A_anno) & reciprocal_exists == "TRUE" & DomainRetainedGene1A == "No" |
        # Also keep select the in-frame reciprocals which have the kinase domain in tact.
        Fusion_Type == "in-frame" & DomainRetainedGene1B == "Yes") %>% 
  # Select to which variables we want here
  select(Sample, FusionName = reciprocal) %>% 
  distinct()
```

Reply:
Ahh
```filter(# Keep the in-frame reciprocals which have the kinase domain in tact.
        grepl("Kinase", Gene1A_anno) & reciprocal_exists == "TRUE" & DomainRetainedGene1A == "No" |
        # Also keep select the in-frame reciprocals which have the kinase domain in tact.
        Fusion_Type == "in-frame" & DomainRetainedGene1B == "Yes")
```
This filtering you have here is filtering on the frame-ness/domain retention of the 5' kinase fusions and not their reciprocals (3' kinase fusions), which is why I was making that variable, then re-joining `five_prime_reciprocals` with the `allFuseLGAT`. 

What I am trying to do is find any 5' kinases without the domain retained, but then go back to that original dataframe `allFuseLGAT` and collect the annotations of their reciprocal fusions (ie 3' kinase fusions' frameness and domain retention), assess whether _those_ are in-frame and retain the kinase domain, then if there is a reciprocal that satisfies that, join _both_ the 5' kinase fusion dataframe (domain lost) and 3' kinase fusion dataframe (inframe/domain retained).

Actually, I can remove:
```
# Add back the reciprocal of the reciprocal (the 5' kinase fusion) and keep this list
five_prime_kinase_keep <- five_prime_reciprocals %>%
  mutate(reciprocal = paste(Gene1B, Gene1A, sep ="--")) %>%
  select(Sample, FusionName = reciprocal) %>%
  distinct()
```

and use:
```
# Then, select the in-frame reciprocals which have the kinase domain in tact. Retain 5' kinase fusion information and update 3' fusion column name to FusionName for merging with allFuseLGAT.
five_prime_kinase_keep <- five_prime_domain_lost %>% 
  select(Sample, five_prime_kinase = FusionName, FusionName = three_prime_kinase) %>%
  left_join(allFuseLGAT, by = c("Sample", "FusionName")) %>%
  filter(Fusion_Type == "in-frame" & DomainRetainedGene1B == "Yes") %>%
  select(Sample, FusionName = five_prime_kinase) %>%
  distinct()
```

`distinct()` is needed because in some cases, there are multiple breakpoints for each fusion and sometimes multiple can have the same frame-ness, so this reduces them to one fusion call per sample.

Reply:
updated with [commit](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/830/commits/ec13333862792e8b02e5726987c5156aeec33666) - @cansavvy let me know if this clarifies!

Reply:
Okay! Thanks for those updates! I think this looks good!